probert/filesystem.py

```diff
@@ -34,7 +34,7 @@ def probe(context=None):
     for device in context.list_devices(subsystem='block'):
         # Ignore block major=1 (ramdisk) and major=7 (loopback)
         # these won't ever be used in recreating storage on target systems.
-        if device['MAJOR'] not in ["1", "7"]:
+        if device.get('MAJOR') not in ["1", "7", None]:
```

Comment:
I think it would be clearer if we spelled out what we're doing here; these _are_ two separate checks and I'd prefer if they weren't conflated:

```py
if 'MAJOR' in device and device['MAJOR'] not in ["1", "7"]:
```

Should we be updating the comment above?

Reply:
FYI. The "device" that triggered this bug, seems to be the one below (extract from udevadm info -e): 
```
 P: /devices/pci0000:17/0000:17:00.0/0000:18:00.0/nvme/nvme0/nvme0c0n1
 L: 0
 E: DEVPATH=/devices/pci0000:17/0000:17:00.0/0000:18:00.0/nvme/nvme0/nvme0c0n1
 E: SUBSYSTEM=block
 E: DEVTYPE=disk
 E: USEC_INITIALIZED=762755939
 E: MPATH_SBIN_PATH=/sbin
 E: DM_MULTIPATH_DEVICE_PATH=0
 E: ID_SERIAL_SHORT=S4CCNE0M300015
 E: ID_WWN=eui.344343304d3000150025384500000004
 E: ID_MODEL=SAMSUNG MZPLL3T2HAJQ-00005
 E: ID_REVISION=GPJA0B3Q
 E: ID_SERIAL=SAMSUNG MZPLL3T2HAJQ-00005_S4CCNE0M300015
 E: ID_PATH=pci-0000:18:00.0-nvme-1
 E: ID_PATH_TAG=pci-0000_18_00_0-nvme-1
 E: TAGS=:systemd:
```